dex/order/status.go

```diff
@@ -55,6 +55,9 @@ const (
 	// standing limit orders that were matched but have failed to swap (neither
 	// executed nor canceled), and preimage misses.
 	OrderStatusRevoked
+
+	// OrderStatusPending is for queued orders.
+	OrderStatusPending
```

Comment:
We (unfortunately) do a lot of inequalities when deciding about order status.  e.g. of `if ord.Status > order.OrderStatusBooked { /* order is done */ }`.

However, prepending a new status is no good because that changes the value of all the other status and the orders stored in the DB are all different statuses effectively, and we absolutely do not want to solve that with an enormous DB upgrade that patches all existing order statuses.

Need to think about this more, but maybe we just root out all inequality based comparisons in favor of explicitly naming status. and make docs clear about the perils of doing that.

Grep the code for:
- `>= order.OrderStat`
- `> order.OrderStat`
- `<= order.OrderStat`
- `< order.OrderStat`

Reply:
I think we should consider a different approach that does not involve a new status defined here (which wouldn't have any meaning for server BTW).  The "pending" nature of the order could be **inferred** by other OrderMetaData fields, or perhaps just by the lack of the real ID and instead only a TempID.

Reply:
> That logic sounds simple enough for frontend (isPending = !o.id && o.tempid or whatever is proper)

I think this is beter too.